File: messmate/backend/src/main/java/com/example/entities/FoodItem.java
package com.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "food_item")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class FoodItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="food_item_id")
    private Integer foodItemId;

    @Column(name="food_name")
    private String foodName;
    
    @Column(name="description")
    private String description;

    @ManyToOne
    @JoinColumn(name = "sub_category_id")
    private SubCategory subCategory;

}

File: backend/Customer/Customer/Models/FoodItem.cs
using System;
using System.Collections.Generic;

namespace Customer.Models;

public partial class FoodItem
{
    public int FoodItemId { get; set; }

    public int SubCategoryId { get; set; }

    public string? FoodName { get; set; }

    public string? Description { get; set; }

    public virtual ICollection<MealmenuFooditem> MealmenuFooditems { get; set; } = new List<MealmenuFooditem>();

    public virtual SubCategory SubCategory { get; set; } = null!;
}
